wakuv2/waku.go

```diff
@@ -1104,10 +1106,11 @@ func (w *Waku) Start() error {
 			w.logger)
 
 		w.missingMsgVerifier.Start(w.ctx)
+		w.logger.Info("Started missing message verifier")
 
 		w.wg.Add(1)
 		go func() {
-			w.wg.Done()
+			defer w.wg.Done()
```

Comment:
not sure how this wait group is intended to behave cc @richard-ramos

Reply:
Adding the defer is correct. This is a bug i introduced when doing the refactoring